### ISSUE TITLE
Building Error

### ISSUE THREAD

**Issue by ptorrente** [Tudat/tudat#498]
I cannot run a single C++ file anymore and I have no idea what is happening. Everything worked, questions 1 and 2 for Assignment 3 worked, and when I tried to run question 3 this error occurred, and now it appears every time, I cannot even run my previous C++ file for Assignment 2.

Any help would be appreciated.

In file included from C:/tudatBundle/eigen/Eigen/Core:347:0,
                 from C:/tudatBundle/tudat/Tudat/Astrodynamics/BasicAstrodynamics/keplerPropagator.h:17,
                 from C:/tudatBundle/tudat/Tudat/SimulationSetup/tudatSimulationHeader.h:17,
                 from C:/tudatBundle/tudat/Tudat/SimulationSetup/tudatEstimationHeader.h:15,
                 from C:\tudatBundle\tudatApplications\NumericalAstrodynamics2018Assignment1\src3\delfiVariationalEquations.cpp:11:
C:/tudatBundle/eigen/Eigen/src/Core/PlainObjectBase.h: In instantiation of 'void Eigen::PlainObjectBase<Derived>::_init2(Eigen::Index, Eigen::Index, typename Eigen::internal::enable_if<(typename Eigen::internal::dense_xpr_base<Derived>::type:: SizeAtCompileTime != 2), T0>::type*) [with T0 = double; T1 = double; Derived = Eigen::Matrix<double, -1, 1>; Eigen::Index = long long int; typename Eigen::internal::enable_if<(typename Eigen::internal::dense_xpr_base<Derived>::type:: SizeAtCompileTime != 2), T0>::type = double]':
C:/tudatBundle/eigen/Eigen/src/Core/Matrix.h:304:35:   required from 'Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols>::Matrix(const T0&, const T1&) [with T0 = double; T1 = double; _Scalar = double; int _Rows = -1; int _Cols = 1; int _Options = 0; int _MaxRows = -1; int _MaxCols = 1]'
C:\tudatBundle\tudatApplications\NumericalAstrodynamics2018Assignment1\src3\delfiVariationalEquations.cpp:413:81:   required from here
C:/tudatBundle/eigen/Eigen/src/Core/PlainObjectBase.h:736:7: error: static assertion failed: FLOATING_POINT_ARGUMENT_PASSED__INTEGER_WAS_EXPECTED
       EIGEN_STATIC_ASSERT(bool(NumTraits<T0>::IsInteger) &&
       ^
mingw32-make.exe[3]: *** [tudatApplications\NumericalAstrodynamics2018Assignment1\src3\CMakeFiles\application_DelfiVariationalEquations.dir\build.make:63: tudatApplications/NumericalAstrodynamics2018Assignment1/src3/CMakeFiles/application_DelfiVariationalEquations.dir/delfiVariationalEquations.cpp.obj] Error 1
mingw32-make.exe[2]: *** [CMakeFiles\Makefile2:17377: tudatApplications/NumericalAstrodynamics2018Assignment1/src3/CMakeFiles/application_DelfiVariationalEquations.dir/all] Error 2
mingw32-make.exe[1]: *** [CMakeFiles\Makefile2:17389: tudatApplications/NumericalAstrodynamics2018Assignment1/src3/CMakeFiles/application_DelfiVariationalEquations.dir/rule] Error 2
mingw32-make.exe: *** [Makefile:4165: application_DelfiVariationalEquations] Error 2
22:49:27: The process "C:\Program Files (x86)\CMake\bin\cmake.exe" exited with code 2.
Error while building/deploying project TudatBundle (kit: Desktop)
When executing step "CMake Build"

**Comment by DominicDirkx** [comment]
A few questions:
- Can you still run example applications or a unit test?
- Could you post a link to your code on Github?

**Comment by ptorrente** [comment]
I ran all the units tests and got this:

99% tests passed, 2 tests failed out of 247

Total Test time (real) = 649.66 sec

The following tests FAILED:
	131 - test_FullPropagationRestrictedThreeBodyProblem (Not Run)
	188 - test_UnscentedKalmanFilter (Failed)
Errors while running CTest
mingw32-make.exe: *** [Makefile:72: test] Error 8
15:04:40: The process "C:\Program Files (x86)\CMake\bin\cmake.exe" exited with code 2.
Error while building/deploying project TudatBundle (kit: Desktop)
When executing step "CMake Build"

The test_UnscentedKalmanFilter fail is normal I know, but I did not have the other one before.

I put my code for Assignment 3 here https://github.com/ptorrente/NumericalAstrodynamics2018Assignment1/tree/master/Assignment%203
I am sure that questions 1 and 2 work, question 3 is still in progress and quite messy, I am pretty sure it won't work as it is.

I should also mention that Qt asked for an update recently, which I did. I can't remember if I was able to run files after that or if I was only working on MATLAB after I created the output files for questions 1 and 2... It seems that I have now Qt Creator 4.8.0 
I have no idea if this could have an impact of any kind but I thought I should mention it.
